src/Workspaces/CSharp/Portable/CodeGeneration/CSharpSyntaxGenerator.cs

```diff
@@ -1202,21 +1202,31 @@ public override IReadOnlyList<SyntaxNode> GetNamespaceImports(SyntaxNode declara
 
         public override SyntaxNode InsertNamespaceImports(SyntaxNode declaration, int index, IEnumerable<SyntaxNode> imports)
```

Comment:
I meant making either making a function like PreserveBanner(SyntaxNode decl, Func<SyntaxNode, SyntaxNode> nodeChanger) or modify PreserveTrivia function to understand the banner..  If you modify PreserveTrivia then you get the behavior correctly everywhere.   Then you would change InsertNamespaceImports to call PreserveTrivia, which would then have an argument to another method InsertNamespaceImportsInternal, following the pattern of other Insert and With APIs.